web/src/App.tsx

```diff
@@ -31,20 +33,57 @@ const TimeControllerWrapper = lazy(() => import('features/time/TimeControllerWra
 
 const isProduction = import.meta.env.PROD;
 
-if (isProduction) {
-  trackEvent(TrackEvent.APP_LOADED, {
-    isNative: Capacitor.isNativePlatform(),
-    platform: Capacitor.getPlatform(),
-  });
+function HandleLegacyRoutes() {
+  const [searchParameters] = useSearchParams();
+
+  const page = (searchParameters.get('page') || 'map')
+    .replace('country', 'zone')
+    .replace('highscore', 'ranking');
+  searchParameters.delete('page');
+
+  const zoneId = searchParameters.get('countryCode');
+  searchParameters.delete('countryCode');
+
+  return (
+    <Navigate
+      to={{
+        pathname: zoneId ? `/zone/${zoneId}` : `/${page}`,
+        search: searchParameters.toString(),
+      }}
+    />
+  );
+}
+
+function ValidZoneIdGuardWrapper({ children }: { children: JSX.Element }) {
+  const [searchParameters] = useSearchParams();
+  const { zoneId } = useParams();
+
+  if (!zoneId) {
+    return <Navigate to="/" replace />;
+  }
+  const upperCaseZoneId = zoneId.toUpperCase();
+  if (zoneId !== upperCaseZoneId) {
+    return <Navigate to={`/zone/${upperCaseZoneId}?${searchParameters}`} replace />;
+  }
+
+  // Handle legacy Australia zone names
+  if (upperCaseZoneId.startsWith('AUS')) {
+    return (
+      <Navigate to={`/zone/${zoneId.replace('AUS', 'AU')}?${searchParameters}`} replace />
+    );
+  }
```

Comment:
Just going to ask this while we are at it:

Should we move this to a redirect rule in Cloudflare instead?